Algorithm/SANGJUN/13305.py

```diff
@@ -0,0 +1,29 @@
+from sys import stdin
+input = stdin.readline
+
+# 현재 도시보다 기름 가격이 싼 도시에 도착하기 전까지 거리만큼 기름 충전
+def _13305(city : int, road : list, price : list):
+    minprice = price[0]
+    min_idx = 0
+    idx = 1
+    ans = 0
+    
+    while idx != city:
+        if minprice > price[idx]:   # 가격이 싼 도시를 찾으면
+            ans += sum(road[min_idx:idx]) * minprice   # 거리만큼의 기름 값
+            minprice, min_idx = price[idx], idx   # 가격정보 갱신
+            idx += 1   # idx 1 증가
+        else:
+            idx += 1
+        
+        if idx == city:   # 마지막 도시에서는 기름을 충전할 일이 없기 때문에 그냥 계산
+            ans += sum(road[min_idx:idx]) * minprice
+            
+    return ans
```

Comment:
```suggestion
    return ans + sum(road[min_idx:idx]) * minprice
```
매번 idx == city를 검사하는 것보다 마지막 answer에 더해주는 것이 더 좋습니다.

Reply:
상민님 코드 보고 깨달았습니다 ㅋㅋㅋ